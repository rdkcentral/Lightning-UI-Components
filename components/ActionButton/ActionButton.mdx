import { Preview, Story } from '@storybook/addon-docs/blocks';

# ActionButton

General purpose button component

## Usage

The basic component just requires a title:

```js
import { ActionButton } from '@lightning/ui';

class Basic extends lng.Component {
  static _template() {
    return {
      ActionButton: {
        type: ActionButton,
        title: 'My Button'
      }
    };
  }
};
```

<Preview>
  <Story id="actionbutton--basic" />
</Preview>

Before the button recieves its title, it should be in a loading state.
<Preview>
  <Story id="actionbutton--loading" />
</Preview>


When focused, the background and text colors are inverted.

<Preview>
  <Story id="actionbutton--focused" />
</Preview>

When and icon is present, the text should shift and all of the content should remain centered in the button.
<Preview>
  <Story id="actionbutton--icon" />
</Preview>

## API

### Properties

name|type|readonly|description
--|--|--|--
-|-|-|-
announce|string|true|announcement text for screen readers
h|number|false|height of the button
title|string|false|button text
unfocusIconUrl|string|false|URI for icon that may live next to the title
w|number|false|width of the button
