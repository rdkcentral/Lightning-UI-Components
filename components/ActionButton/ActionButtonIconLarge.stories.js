import lng from '@lightningjs/core';

import { ActionButtonIconLarge } from '.';
import mdx from './ActionButtonIconLarge.mdx';
import icon from '../../assets/images/ic_lightning_white_32.png';

export default {
  title: 'ActionButtonIconLarge',
  parameters: {
    tag: 'ActionButton',
    docs: {
      page: mdx
    }
  }
};

const args = { icon };
const argTypes = {
  backgroundType: {
    control: {
      type: 'radio',
      options: ['stroke', 'fill', 'float', 'ghost']
    }
  },
  focused: { control: 'boolean' },
  onEnter: { action: 'onEnter' }
};
const parameters = {
  argActions: {
    focused: (isFocused, component) => {
      component._getFocused = isFocused
        ? () => component.tag('ActionButton')
        : () => {};
      component._refocus();
    }
  }
};

export const Base = args =>
  class Base extends lng.Component {
    static _template() {
      return {
        ActionButton: {
          type: ActionButtonIconLarge,
          icon: args.icon,
          backgroundType: args.backgroundType,
          onEnter: args.onEnter
        }
      };
    }

    _getFocused() {
      if (args.focused) {
        return this.tag('ActionButton');
      }
    }
  };
Base.args = args;
Base.argTypes = argTypes;
Base.parameters = parameters;

export const Loading = () =>
  class Loading extends lng.Component {
    static _template() {
      return {
        ActionButton: {
          type: ActionButtonIconLarge
        }
      };
    }
  };

Loading.args = null;
