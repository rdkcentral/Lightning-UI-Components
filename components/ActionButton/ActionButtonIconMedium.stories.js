import lng from 'wpe-lightning';

import { ActionButtonIconMedium } from '.';
import mdx from './ActionButtonIconMedium.mdx';
import icon from '../../assets/images/ic_lightning_white_32.png';

export default {
  title: 'ActionButtonIconMedium',
  parameters: {
    tag: 'ActionButton',
    docs: {
      page: mdx
    }
  }
};

const args = { icon };
const argTypes = {
  backgroundType: {
    control: {
      type: 'radio',
      options: ['stroke', 'fill', 'float', 'ghost']
    }
  },
  focused: { control: 'boolean' },
  onEnter: { action: 'onEnter' }
};
const parameters = {
  argActions: {
    focused: (isFocused, component) => {
      component._getFocused = isFocused
        ? () => component.tag('ActionButton')
        : () => {};
      component._refocus();
    }
  }
};

export const Base = args =>
  class Base extends lng.Component {
    static _template() {
      return {
        ActionButton: {
          type: ActionButtonIconMedium,
          icon: args.icon,
          backgroundType: args.backgroundType,
          onEnter: args.onEnter
        }
      };
    }

    _getFocused() {
      if (args.focused) {
        return this.tag('ActionButton');
      }
    }
  };
Base.args = args;
Base.argTypes = argTypes;
Base.parameters = parameters;

export const Loading = () =>
  class Loading extends lng.Component {
    static _template() {
      return {
        ActionButton: {
          type: ActionButtonIconMedium
        }
      };
    }
  };

Loading.args = null;
