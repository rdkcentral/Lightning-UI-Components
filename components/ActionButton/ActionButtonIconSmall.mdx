import { Canvas, Story } from '@storybook/addon-docs/blocks';
import { Theme } from '../../.storybook/utils';
import { ActionButtonIconSmall } from '.';

# ActionButtonIconSmall

Small [ActionButton](?path=/docs/actionbutton--base) for a single icon.

## Usage

The basic component just requires a icon:

```js
import { ActionButtonIconSmall } from '@lightning/ui';
// or
// import { ActionButtonIconSmall } from '@lightning/ui/components/ActionButton';

class Basic extends lng.Component {
  static _template() {
    return {
      ActionButton: {
        type: ActionButtonIconSmall,
        icon: 'path/to/icon',
        backgroundType: 'stroke'
      }
    };
  }
};
```

The expected icon size is 40px by 40px. See [Icon](?path=/docs/icon--basic) for options.

<Canvas>
  <Story id="actionbuttoniconsmall--base" />
</Canvas>


Before the Action Button receives its icon, it should be in a loading state.
<Canvas>
  <Story id="actionbuttoniconsmall--loading" />
</Canvas>

## API

### Properties

name|type|required|default|description
--|--|--|--|--
announce|string|true|label or title|announcement text for screen readers
backgroundType|'stroke'\|'fill'\|'float'\|'ghost'|false|'float'|color of `Button` container.
fixed|boolean|false|true|when true, `w` overrides `minWidth`
h|number|false|64|height
icon|string|false|-|URI for icon or inline svg that may live next to the title. See <a href="?path=/docs/icon--basic">Icon</a>
padding|number|false|16|left and right padding value
radius|number|false|8|border radius
w|number|false|64|width

### Methods

#### onEnter(thisArg): void

Callback to be executed on <kbd>Enter</kbd> key down event

### Styles

Styled components can be adjusted by composing using [withStyles](?path=/docs/mixins-withstyles--basic). Below is the default `styles` object

<Theme theme={ActionButtonIconSmall.styles} />