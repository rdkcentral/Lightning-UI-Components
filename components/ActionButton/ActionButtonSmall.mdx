import { Canvas, Story } from '@storybook/addon-docs/blocks';

# ActionButtonSmall

Small [ActionButton](?path=/docs/actionbutton--base).

## Usage

The basic component just requires a title:

```js
import { ActionButtonSmall } from '@lightning/ui';
// or
// import { ActionButtonSmall } from '@lightning/ui/components/ActionButton';

class Basic extends lng.Component {
  static _template() {
    return {
      ActionButton: {
        type: ActionButtonSmall,
        title: 'Action Button',
        backgroundType: 'stroke'
      }
    };
  }
};
```

An `icon` can also be included. The expected icon size is 32px by 32px. See [Icon](?path=/docs/icon--basic) for options.

<Canvas>
  <Story id="actionbuttonsmall--base" />
</Canvas>


Before the Action Button receives its title, it should be in a loading state.
<Canvas>
  <Story id="actionbuttonsmall--loading" />
</Canvas>

## API

### Properties

name|type|required|default|description
--|--|--|--|--
announce|string|true|label or title|announcement text for screen readers
backgroundType|'stroke'\|'fill'\|'float'\|'ghost'|false|'float'|color of `Button` container.
fixed|boolean|false|false|when true, `w` overrides `minWidth`
h|number|false|64|height
icon|string|false|-|URI for icon or inline svg that may live next to the title. See <a href="?path=/docs/icon--basic">Icon</a>
minWidth|number|false|185|minimum width of the button
padding|number|false|16|left and right padding value
radius|number|false|8|border radius
title|string|false|-|button text
w|number|false|-|width

### Methods

#### onEnter(thisArg): void

Callback to be executed on <kbd>Enter</kbd> key down event
