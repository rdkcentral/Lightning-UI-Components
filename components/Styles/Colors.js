/**
 * Colors
 *
 * Contains global color style information to easily maintain consistency throughout components.
 */
import tokens from 'xfinity-ds-tokens/dist/lightning-ui-tokens/lightning/lightning-ui';
import { rgba2argb } from '../../utils';

/**
 * Combines rgb hex string and alpha into argb hexadecimal number
 * @param {string|number} hex - 6 alphanumeric characters between 0-f or argb hexadecimal number
 * @param {number} [alpha] - number between 0-100 (0 is invisible, 100 is opaque)
 */
export function getHexColor(hex, alpha = 100) {
  if (!hex) {
    return 0x00;
  }

  if (typeof hex === 'number') {
    hex = hex.toString(16).slice(2);
  }

  hex = hex.replace('#', '');

  let hexAlpha = Math.round((alpha / 100) * 255).toString(16);
  let str = `0x${hexAlpha}${hex}`;
  return Number(str);
}

/**
 * Returns valid string of HEX color
 *
 * @param {string} color
 * @param {boolean} fill
 */
export function getValidColor(color) {
  if (/^0x[0-9a-fA-F]{8}/g.test(color)) {
    // User enters a valid 0x00000000 hex code
    return Number(color);
  } else if (/^#[0-9a-fA-F]{6}/g.test(color)) {
    // User enters valid #000000 hex code
    return getHexColor(color.substr(1, 6));
  } else if (typeof color === 'string' && /^[0-9]{8,10}/g.test(color)) {
    return parseInt(color);
  } else if (
    typeof color === 'number' &&
    /^[0-9]{8,10}/g.test(color.toString())
  ) {
    return color;
  } else if (typeof color === 'string' && color.indexOf('rgba') > -1) {
    return rgba2argb(color);
  }
  return null;
}

/**
 * Xfinity-branded color palette for Lightning
 * https://share.goabstract.com/06d0601d-b7d2-48f3-93a2-17cc78bedb5e?mode=build&sha=6d9d06ec3e924eb83ff133826956e1e9a4fc7e36
 */
export const GREY = {
  0: getHexColor(tokens.palette.white),
  5: getHexColor(tokens.palette.grey05),
  10: getHexColor(tokens.palette.grey10),
  20: getHexColor(tokens.palette.grey20),
  30: getHexColor(tokens.palette.grey30),
  40: getHexColor(tokens.palette.grey40),
  50: getHexColor(tokens.palette.grey50),
  60: getHexColor(tokens.palette.grey60),
  70: getHexColor(tokens.palette.grey70),
  80: getHexColor(tokens.palette.grey80),
  90: getHexColor(tokens.palette.grey90),
  100: getHexColor(tokens.palette.black),
  __isColor: true
};

export const BLUE = {
  hover: getHexColor(tokens.palette.blue10),
  default: getHexColor(tokens.palette.blue20),
  pressed: getHexColor(tokens.palette.blue30),
  indicator: getHexColor(tokens.palette.blue40),
  shade: getHexColor(tokens.palette.blue90),
  __isColor: true
};

export const GREEN = {
  hover: getHexColor(tokens.palette.green10),
  default: getHexColor(tokens.palette.green20),
  pressed: getHexColor(tokens.palette.green30),
  indicator: getHexColor(tokens.palette.green40),
  shade: getHexColor(tokens.palette.green90),
  __isColor: true
};

export const YELLOW = {
  hover: getHexColor(tokens.palette.yellow10),
  default: getHexColor(tokens.palette.yellow20),
  pressed: getHexColor(tokens.palette.yellow30),
  indicator: getHexColor(tokens.palette.yellow40),
  shade: getHexColor(tokens.palette.yellow90),
  __isColor: true
};

export const RED = {
  hover: getHexColor(tokens.palette.red10),
  default: getHexColor(tokens.palette.red20),
  pressed: getHexColor(tokens.palette.red30),
  indicator: getHexColor(tokens.palette.red40),
  shade: getHexColor(tokens.palette.red90),
  __isColor: true
};

export const PURPLE = {
  base: getHexColor(tokens.palette.purple50),
  hover: getHexColor(tokens.palette.purple10),
  default: getHexColor(tokens.palette.purple20),
  pressed: getHexColor(tokens.palette.purple30),
  indicator: getHexColor(tokens.palette.purple40),
  shade: getHexColor(tokens.palette.purple90),
  __isColor: true
};

export const OPACITY = {
  0: 0,
  1: 16,
  2: 32,
  3: 48,
  4: 64,
  5: 80,
  6: 96
};

export const PALETTE = {
  getHexColor,
  black: GREY[100],
  white: GREY[0],
  grey: GREY,
  blue: BLUE,
  green: GREEN,
  yellow: YELLOW,
  red: RED,
  purple: PURPLE,
  opacity: OPACITY,
  __isColor: true,
  background: {
    default: GREY[80],
    stroke: 0x00,
    float: 0x00,
    fill: GREY[70],
    focus: GREY[10],
    ghost: getHexColor(GREY[70], 48),
    __isColor: true
  },
  text: {
    light: {
      primary: getHexColor(GREY[0], 95),
      secondary: getHexColor(GREY[0], 80),
      tertiary: getHexColor(GREY[0], 60),
      __isColor: true
    },
    dark: {
      primary: getHexColor(GREY[100], 95),
      secondary: getHexColor(GREY[100], 80),
      tertiary: getHexColor(GREY[100], 60),
      __isColor: true
    },
    focus: getHexColor(GREY[100], 95),
    __isColor: true
  },
  badge: {
    default: getHexColor(GREY[10], 32),
    __isColor: true
  },
  focusRing: {
    primary: getHexColor(GREY[10], 96),
    secondary: getHexColor(GREY[0], 8),
    __isColor: true
  }
};
