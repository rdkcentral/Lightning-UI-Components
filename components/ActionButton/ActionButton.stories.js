import lng from 'wpe-lightning';
import { withKnobs, text, boolean, radios } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import ActionButton from '.';
import mdx from './ActionButton.mdx';
import icon from '../../assets/images/ic_lightning_white_32.png';

export default {
  title: 'ActionButton',
  component: ActionButton,
  decorators: [withKnobs],
  parameters: {
    docs: {
      page: mdx
    }
  }
};

const makeOptions = (...opts) => {
  return opts.reduce(
    (obj, key) => ({
      ...obj,
      [key]: key.toLowerCase()
    }),
    {}
  );
};

export const Base = () =>
  class Base extends lng.Component {
    static _template() {
      return {
        ActionButton: {
          type: ActionButton,
          backgroundType: radios(
            'Background',
            makeOptions('stroke', 'fill', 'float', 'ghost'),
            'stroke'
          ),
          title: text('Title', 'Type Something'),
          onEnter: action('onEnter')
        }
      };
    }
    _init() {
      if (boolean('Icon', false)) {
        this.tag('ActionButton').icon = icon;
      }
    }

    _getFocused() {
      if (boolean('Focused', false)) {
        return this.tag('ActionButton');
      }
    }
  };

export const Loading = () =>
  class Loading extends lng.Component {
    static _template() {
      return {
        ActionButton: {
          type: ActionButton
        }
      };
    }
  };
