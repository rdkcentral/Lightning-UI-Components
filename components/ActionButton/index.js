/**
 * Action Button Component
 *
 * Action Button Class that contains styling and functionality for all Action Button items
 */
import lng from 'wpe-lightning';
import Button from '../Button';
import withStyles from '../../mixins/withStyles';
import { RoundRect } from '../../utils';

// styles
import { getHexColor, getFocusScale } from '../Styles';

export const styles = theme => ({
  minWidth: 410,
  h: 72,
  background: {
    color: theme.palette.background.float
  },
  backgrounds: theme.palette.background,
  radius: theme.border.radius.small,
  text: {
    ...theme.typography.button1,
    color: theme.palette.text.light.primary
  },
  padding: theme.spacing(2),
  shadow: theme.materials.glow,
  icon: {
    color: theme.palette.text.light.primary,
    size: theme.sizes.icon.medium,
    spacing: theme.spacing(1)
  },
  loading: {
    color: getHexColor('ffffff', 24)
  },
  stroke: {
    weight: 2,
    color: theme.palette.grey[5]
  },
  focused: {
    background: { color: theme.palette.background.focus },
    icon: { color: theme.palette.text.dark.primary },
    text: { color: theme.palette.text.dark.primary }
  }
});

class ActionButtonBase extends Button {
  static _template() {
    return {
      ...super._template(),
      minWidth: this.styles.minWidth,
      h: this.styles.h,
      padding: this.styles.padding,
      radius: this.styles.radius,
      text: this.styles.text,
      focus: this.styles.focus,
      unfocus: this.styles.unfocus,
      backgroundType: 'float',
      Loader: {
        color: this.styles.loading.color
      },
      DropShadow: {
        alpha: 0
      }
    };
  }

  _setIcon(src) {
    return src ? { ...this.styles.icon, src } : null;
  }

  _updateDropShadow() {
    const DropShadow = this.styles.shadow({
      w: this.w,
      h: this.h,
      borderRadius: this.styles.radius
    });
    this._DropShadow.patch(DropShadow);
    const alpha = Number(this._focused);
    if (this._smooth) {
      this._DropShadow.smooth = { alpha };
    } else {
      this._DropShadow.alpha = alpha;
    }
  }

  _updateScale() {
    const scale = this._focused ? getFocusScale(this.w) : 1;
    if (this._smooth) {
      this.smooth = { scale };
    } else {
      this.scale = scale;
    }
  }

  _updateLoader() {
    if (!(this.title || this.icon)) {
      this.patch({ color: 0x00 });
      this._Title.patch({ texture: false });
      this._Stroke.patch({ texture: false });
      this._Loader.patch({
        texture: lng.Tools.getRoundRect(
          RoundRect.getWidth(this.w),
          RoundRect.getHeight(this.h - 2),
          this.styles.radius
        )
      });
      if (this._loading) this._loading.start();
    } else {
      this._Loader.patch({ texture: false });
      if (this._loading) this._loading.stop();
    }
  }

  _update() {
    if (this._readyForUpdates) {
      this._updateDropShadow();
      this._updateScale();
      this._updateLoader();
      if (this.title || this.icon) super._update();
    }
  }

  _firstEnable() {
    this._loading = this._Loader.animation({
      duration: 2,
      repeat: -1,
      stopMethod: 'immediate',
      actions: [{ p: 'alpha', v: { 0: 0.5, 0.5: 1, 1: 0.5 } }]
    });
    this._loading.start();
    super._firstEnable && super._firstEnable();
  }

  get backgroundType() {
    return this._backgroundType;
  }

  set backgroundType(backgroundType) {
    if (this._backgroundType !== backgroundType) {
      this._backgroundType = backgroundType;
      if (this.styles) {
        const background =
          this.styles.backgrounds[backgroundType] ||
          this.styles.backgrounds.float;
        this.background = background;
        this.stroke = backgroundType === 'stroke';

        if (this.stroke) {
          this.strokeWeight = this.styles.stroke.weight;
          this.strokeColor = this.styles.stroke.color;
        }
      }
      this._update();
    }
  }

  get _Loader() {
    return this.tag('Loader');
  }
  get _DropShadow() {
    return this.tag('DropShadow');
  }
}

class ActionButton extends withStyles(ActionButtonBase, styles) {}

export default ActionButton;

export const largeStyles = theme => ({
  h: 96,
  minWidth: 320,
  text: theme.typography.headline1,
  icon: {
    size: theme.sizes.icon.large
  }
});

export const ActionButtonLarge = withStyles(
  class ActionButtonLarge extends ActionButton {},
  largeStyles
);

export const iconLargeStyles = () => ({
  h: 96,
  w: 96
});

export const ActionButtonIconLarge = withStyles(
  class ActionButtonIconLarge extends ActionButtonLarge {
    _construct() {
      super._construct();
      this._fixed = true;
    }
  },
  iconLargeStyles
);

export const ActionButtonMedium = class ActionButtonMedium extends ActionButton {};

export const iconMediumStyles = () => ({
  h: 72,
  w: 72
});

export const ActionButtonIconMedium = withStyles(
  class ActionButtonIconMedium extends ActionButtonMedium {
    _construct() {
      super._construct();
      this._fixed = true;
    }
  },
  iconMediumStyles
);

export const smallStyles = theme => ({
  h: 64,
  minWidth: 185,
  text: theme.typography.button2,
  icon: {
    size: theme.sizes.icon.small
  }
});

export const ActionButtonSmall = withStyles(
  class ActionButtonSmall extends ActionButton {},
  smallStyles
);

export const iconSmallStyles = () => ({
  h: 64,
  w: 64
});

export const ActionButtonIconSmall = withStyles(
  class ActionButtonIconSmall extends ActionButtonSmall {
    _construct() {
      super._construct();
      this._fixed = true;
    }
  },
  iconSmallStyles
);

export const xSmallStyles = theme => ({
  h: 48,
  minWidth: 152,
  text: theme.typography.button2,
  icon: {
    size: theme.sizes.icon.small
  }
});

export const ActionButtonXsmall = withStyles(
  class ActionButtonXsmall extends ActionButton {},
  xSmallStyles
);
