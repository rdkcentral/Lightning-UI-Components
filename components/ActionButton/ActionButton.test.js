import ActionButton from '.';
import TestRenderer from '../lightning-test-renderer';
// styles
import { getHexColor, COLORS_NEUTRAL, COLORS_TEXT } from '../Styles';

const Component = {
  Component: {
    type: ActionButton,
    title: 'Button'
  }
};

describe('ActionButton', () => {
  let testRenderer, button;

  beforeEach(() => {
    testRenderer = TestRenderer.create(Component);
    button = testRenderer.getInstance();
    return button._whenEnabled;
  });

  it('should render', () => {
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should load before recieving data', () => {
    testRenderer = TestRenderer.create({ Component: { type: ActionButton } });
    button = testRenderer.getInstance();
    expect(button._loading.isPlaying()).toBe(true);
  });

  it('should append Button to end of announce context', () => {
    expect(button.announce).toEqual('Button, Button');
  });

  it('should patch in an icon if provided', () => {
    testRenderer = TestRenderer.create({
      Component: { type: ActionButton, title: 'Button', icon: '/path/to/icon' }
    });
    button = testRenderer.getInstance();
    expect(button.icon).toEqual('/path/to/icon');
    expect(button._Icon).toBeDefined();
  });

  it('should update button properties when focused', done => {
    button._focus();
    testRenderer.update();
    let focusColors = {
      text: getHexColor(COLORS_TEXT.dark),
      background: getHexColor(COLORS_NEUTRAL.light2)
    };
    button._whenEnabled.then(() => {
      expect(button._Background.transition('color').targetValue).toBe(
        focusColors.background
      );
      expect(button._Title.transition('color').targetValue).toBe(
        focusColors.text
      );
      expect(button._DropShadow.transition('alpha').targetValue).toBe(1);
      done();
    });
  });

  it('should update properties when unfocused', done => {
    button._unfocus();
    testRenderer.update();
    let focusColors = {
      text: getHexColor(COLORS_TEXT.light),
      background: 0
    };
    button._whenEnabled.then(() => {
      expect(button._Background.transition('color').targetValue).toBe(
        focusColors.background
      );
      expect(button._Title.transition('color').targetValue).toBe(
        focusColors.text
      );
      expect(button._DropShadow.transition('alpha').targetValue).toBe(0);
      done();
    });
  });
});
