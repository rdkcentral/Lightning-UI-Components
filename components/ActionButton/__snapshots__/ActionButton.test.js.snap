// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton should render 1`] = `
Object {
  "Component": Object {
    "Button": Object {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": Object {
        "Background": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": "0",
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "id": 8,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Background",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Background",
          ],
          "texture": Object {
            "type": "StaticCanvasTexture",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
        "Content": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": Object {
            "Icon": Object {
              "active": false,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 0,
              "id": 5,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Icon",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": Array [
                "Icon",
              ],
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 8,
              "zIndex": 0,
            },
            "Title": Object {
              "active": false,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 0,
              "id": 6,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Title",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": Array [
                "Title",
              ],
              "texture": Object {
                "fontFace": "XfinityStandardBold",
                "fontSize": 28,
                "lineHeight": 36,
                "maxLinesSuffix": "...",
                "text": "Button",
                "textColor": 4294638330,
                "textOverflow": "ellipsis",
                "type": "TextTexture",
              },
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": true,
          "flexItem": true,
          "h": 88,
          "id": 4,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Content",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Content",
          ],
          "visible": true,
          "w": 410,
          "x": 0,
          "y": 0,
          "zIndex": 2,
        },
        "Stroke": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "id": 7,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Stroke",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Stroke",
          ],
          "texture": Object {
            "type": "StaticCanvasTexture",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
      },
      "clipping": false,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 0,
      "id": 3,
      "isComponent": undefined,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "Button",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": undefined,
      "tag": [Function],
      "tags": Array [
        "Button",
      ],
      "visible": true,
      "w": 0,
      "x": 0,
      "y": 0,
      "zIndex": 1,
    },
  },
}
`;
